From: Lennart Weller <lhw@ring0.de>
Date: Wed, 29 Apr 2020 22:11:05 +0200
Subject: Makefile changes for debian packaging

 - Add debian package version to build version for easier debugging
 - Include debian package buildflags
---
 client/CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index ddae7b9..f27b969 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -7,7 +7,7 @@ include(GNUInstallDirs)
 include(CheckCCompilerFlag)
 include(FeatureSummary)
 
-option(OPTIMIZE_FOR_NATIVE "Build with -march=native" ON)
+option(OPTIMIZE_FOR_NATIVE "Build with -march=native" OFF)
 if(OPTIMIZE_FOR_NATIVE)
   CHECK_C_COMPILER_FLAG("-march=native" COMPILER_SUPPORTS_MARCH_NATIVE)
   if(COMPILER_SUPPORTS_MARCH_NATIVE)
@@ -34,9 +34,11 @@ add_compile_options(
   "-ffast-math"
   "-fdata-sections"
   "-ffunction-sections"
-  "$<$<CONFIG:DEBUG>:-O0;-g3;-ggdb>"
 )
 
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} $ENV{CFLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} $ENV{CPPFLAGS}")
+
 set(EXE_FLAGS "-Wl,--gc-sections")
 set(CMAKE_C_STANDARD 11)
 
@@ -59,7 +61,8 @@ execute_process(
 
 find_package(GMP)
 
-add_definitions(-D BUILD_VERSION='"${BUILD_VERSION}"')
+string(CONCAT OUTPUT_VERSION '"${BUILD_VERSION}"' '" - "' '"debian/$ENV{DEB_VERSION}"')
+add_definitions(-D BUILD_VERSION='"${OUTPUT_VERSION}"')
 add_definitions(-D ATOMIC_LOCKING)
 add_definitions(-D GL_GLEXT_PROTOTYPES)
 get_filename_component(PROJECT_TOP "${PROJECT_SOURCE_DIR}/.." ABSOLUTE)
